Automattic/pocket-casts-ios
# Categories: Re-merge pull requests for Categories Redesign

This re-applies all commits from Pull Requests related to the Categories Redesign for the 7.63 release.

I cherry picked the commits from the following PRs:
#1717
#1712
#1660
#1648
#1631
#1626
#1622
#1621
#1620
#1614
#1611
#1605
#1604
#1602
#1570

https://github.com/Automattic/pocket-casts-ios/assets/3250/167f0c4f-384f-4b7b-8d46-cddd147d462c

## To test

All steps in previous PRs can be followed for thorough testing

* Open Discover tab
* Ensure that new Category tabs are shown
* Ensure no sponsored podcasts are shown on the Discover page (outside of the Featured carousel)
* Tap on a category tab
* Ensure a single Sponsored podcast appears here (for most categories)
* Ensure tapping the Sponsored podcast works

## Checklist
